lib/record_store/provider/ns1.rb

```diff
@@ -0,0 +1,192 @@
+require 'ns1'
+
+module RecordStore
+  class Provider::NS1 < Provider
+    class << self
+      def client
+        Provider::NS1::Client.new(api_key: secrets['api_key'])
+      end
+
+      # Downloads all the records from the provider.
+      #
+      # Returns: an array of `Record` for each record in the provider's zone
+      def retrieve_current_records(zone:, stdout: $stdout)
+        records_for_zone(zone)
+         .map{|short_record| build_from_api(short_record, zone)}
+         .flatten
+      end
+
+      # Returns an array of the zones managed by provider as strings
+      def zones
+        client.zones.map do |zone|
+          zone["zone"]
+        end
+      end
+
+      private
+
+      # Fetches simplified records for the provided zone
+      def records_for_zone(zone)
+        client.zone(zone)["records"]
+      end
+
+      # Creates a new record to the zone. It is expected this call modifies external state.
+      #
+      # Arguments:
+      # record - a kind of `Record`
+      def add(record, zone)
+        new_answers = [{ answer: [ build_api_answer_from_record(record) ]}]
+
+        record_fqdn = record.fqdn.sub(/.$/, '')
+
+        existing_record = client.record(
+          zone: zone,
+          fqdn: record_fqdn,
+          type: record.type
+        )
+
+        if existing_record.nil?
+          client.create_record(
+            zone: zone,
+            fqdn: record_fqdn,
+            type: record.type,
+            params: { answers: new_answers, ttl: record.ttl }
+          )
+          return
+        end
+
+        existing_answers = existing_record["answers"].map { |answer| symbolize_keys(answer) }
+        client.modify_record(
+          zone: zone,
+          fqdn: record_fqdn,
+          type: record.type,
+          params: { answers: existing_answers + new_answers }
```

Comment:
No update of the TTL ?
How is the TTL handled with many answers ? (taking the shortest ? the longest ? the first we read ?)

Reply:
I think this questions related to what we are discussing about TTL now. Many answers will be adding to the record with the TTL which was set for the creation of the record. So I could say, the TTL will be handled with the first we read from given questions.